### PR TITLE
Add benchmarks for cross entropy loss

### PR DESCRIPTION
**Pull request by protonu**
This add operator benchmarks for cross entropy loss.

Please note that the computation for IO Bytes is approximate.

### CODE REVIEW

**Comment by protonu** [comment]
To get `take_along_axis` to run via nvFuser, we need to uncomment `take_along_axis` and `take` here:
https://github.com/Lightning-AI/lightning-thunder/blob/70c06666ee8847746f836d178aee541fe749c62f/thunder/executors/nvfuserex_impl.py#L1373-L1378

**Comment by protonu** [comment]
# please note that this computes the IO bytes based on the graph/decomposition
    ## ===== Backward graph 0 =====
    ## <eval_with_key>.3 class GraphModule(torch.nn.Module):
    ## def forward(self, primals_1: "i64[8192][1]cuda:0", primals_2: "f32[8192, 3024][3024, 1]cuda:0",
    #        amax: "f32[8192, 1][1, 1]cuda:0", log: "f32[8192, 1][1, 1]cuda:0", convert_element_type: "f32[][]cuda:0", tangents_1: "f32[][]cuda:0"):
    ##     # File: /opt/pytorch/nvfuser/test.py:6 in fn, code: return torch.nn.functional.cross_entropy(input, target)
    ##    div_1: "f32[][]cuda:0" = torch.ops.aten.div.Tensor(tangents_1, convert_element_type);  tangents_1 = convert_element_type = None
    ##    unsqueeze_1: "i64[8192, 1][1, 1]cuda:0" = torch.ops.aten.unsqueeze.default(primals_1, 1);  primals_1 = None
    ##    ne_3: "b8[8192, 1][1, 1]cuda:0" = torch.ops.aten.ne.Scalar(unsqueeze_1, -100)
    ##    full_default: "i64[][]cuda:0" = torch.ops.aten.full.default([], 0, dtype = torch.int64, layout = torch.strided, device = device(type='cuda', index=0), pin_memory = False)
    ##    where_2: "i64[8192, 1][1, 1]cuda:0" = torch.ops.aten.where.self(ne_3, unsqueeze_1, full_default);  unsqueeze_1 = full_default = None
    ##    full_default_3: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.full.default([8192, 3024], 0, dtype = torch.float32, layout = torch.strided, device = device(type='cuda', index=0), pin_memory = False)
    ##    scatter: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.scatter.value(full_default_3, 1, where_2, -1.0);  full_default_3 = where_2 = None
    ##    full_default_1: "f32[][]cuda:0" = torch.ops.aten.full.default([], 0.0, dtype = torch.float32, layout = torch.strided, device = device(type='cuda', index=0), pin_memory = False)
    ##    where_3: "f32[8192, 1][1, 1]cuda:0" = torch.ops.aten.where.self(ne_3, div_1, full_default_1);  ne_3 = div_1 = full_default_1 = None
    ##    mul: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.mul.Tensor(scatter, where_3);  scatter = where_3 = None
    ##    sub: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.sub.Tensor(primals_2, amax);  primals_2 = amax = None
    ##    sub_1: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.sub.Tensor(sub, log);  sub = log = None
    ##    exp_1: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.exp.default(sub_1);  sub_1 = None
    ##    sum_4: "f32[8192, 1][1, 1]cuda:0" = torch.ops.aten.sum.dim_IntList(mul, [1], True)
    ##    mul_1: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.mul.Tensor(exp_1, sum_4);  exp_1 = sum_4 = None
    ##    sub_2: "f32[8192, 3024][3024, 1]cuda:0" = torch.ops.aten.sub.Tensor(mul, mul_1);  mul = mul_1 = None
    ##    return (None, sub_2)

File: benchmarks/python/cross_entropy_loss.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import torch
+from functools import partial
+
+
+from .model_configs import configs
+
+
+class CrossEntropyLossBase:
+    def __init__(self, model_name, dtype):
+        self.config = configs[model_name]()
```
Comment (jjsjann123) [inline benchmarks/python/cross_entropy_loss.py:14]: Nice~~ you are also re-using the configs :clap:

File: benchmarks/python/cross_entropy_loss.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    def __init__(self, dtype):
+        super().__init__("hf_phi3", dtype)
+
+    def model(self):
+        from transformers.models.phi3 import Phi3PreTrainedModel
+
+        class MyModel(Phi3PreTrainedModel):
+            def __init__(self, config):
+                super().__init__(config)
+                self.padding_idx = config.pad_token_id
```
Comment (jjsjann123) [inline benchmarks/python/cross_entropy_loss.py:116]: self.padding_idx isn't used here.

File: benchmarks/python/cross_entropy_loss.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    def __init__(self, dtype):
+        super().__init__("hf_mistral_nemo", dtype)
+
+    def model(self):
+        from transformers.models.phi3 import Phi3PreTrainedModel
+
+        class MyModel(Phi3PreTrainedModel):
+            def __init__(self, config):
+                super().__init__(config)
+                self.padding_idx = config.pad_token_id
```
Comment (jjsjann123) [inline benchmarks/python/cross_entropy_loss.py:144]: ditto

File: benchmarks/python/cross_entropy_loss.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+
+class HfMistralNemo(CrossEntropyLossBase):
+    def __init__(self, dtype):
+        super().__init__("hf_mistral_nemo", dtype)
+
+    def model(self):
+        from transformers.models.phi3 import Phi3PreTrainedModel
+
+        class MyModel(Phi3PreTrainedModel):
```
Comment (jjsjann123) [inline benchmarks/python/cross_entropy_loss.py:141]: This is supposed to be `MistralPreTrainedModel`.

**Review by jjsjann123 — CHANGES_REQUESTED** [review]

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Comment by protonu** [comment]
!test

**Comment by protonu** [comment]
!test

**Comment by protonu** [comment]
!test

**Comment by protonu** [comment]
!test

**Comment by protonu** [comment]
!test
